src/Database/LSMTree/Normal.hs

```diff
@@ -272,7 +294,7 @@ snapshot ::
   -> m ()
 snapshot = undefined
 
--- | Open a table through a snapshot, returning a new table handle.
+-- | Open a table from a named snapshot, returning a new table handle.
```

Comment:
Addition: 

* Opening a snapshot that doesn't exist should result in an exception.
* If we try to open a snapshot but the types of keys+values+blobs are not as expected, then an exception should be thrown. For example, if a snapshot was created from a `TableHandle m Int Int Int`, but we open it as `TableHandle m Bool Bool Bool`, we get an exception

Reply:
> f we try to open a snapshot but the types of keys+values+blobs are not as expected

How that can be enforced? E.g. `Typeable` is not stable value (if type is moved to the different module, it will change).